osu.Game/Online/API/Requests/ResponseWithCursor.cs

```diff
@@ -11,6 +11,6 @@ public abstract class ResponseWithCursor
         /// A collection of parameters which should be passed to the search endpoint to fetch the next page.
         /// </summary>
         [JsonProperty("cursor")]
-        public dynamic CursorJson;
+        public object CursorJson;
```

Comment:
I had to change this from `dynamic` -> `object` to get this to compile; working with `dynamic` directly in the codebase caused issues. However, this doesn't seem like a great solution to me; maybe `ResponseWithCursor` should be generic w.r.t. the cursor type?

Reply:
I would agree with this due to reasons outlined in the other comment. Maybe worth considering making the generic explicitly opt-in to avoid having to update previous usages (as in having both `ResponseWithCursor` and `ResponseWithCursor<T>`).  I'd also probably argue that cursor classes should be inner classes of the requests they correspond to themselves, unless deemed reusable (in the structural *and* semantic sense)

Ideally we wouldn't have to interpret this at all but alas...